xbmc/pictures/GUIWindowSlideShow.cpp

```diff
@@ -1164,7 +1165,9 @@ void CGUIWindowSlideShow::OnLoadPic(int iPic, int iSlideNumber, const std::strin
 
 void CGUIWindowSlideShow::Shuffle()
 {
-  std::random_shuffle(m_slides.begin(), m_slides.end());
+  std::random_device rng;
+  std::mt19937 urng(rng());
+  std::shuffle(m_slides.begin(), m_slides.end(), urng);
```

Comment:
There is a util in https://github.com/xbmc/xbmc/blob/master/xbmc/utils/Random.h for this because it's a quite common pattern.

Reply:
thanks. didn't know about utility realization.